exec/mailinglist.R

```diff
@@ -51,56 +47,62 @@ if(arguments[["tabulate"]] & arguments[["help"]]){
   conf <- yaml::read_yaml(conf_path)
 
   perceval_path <- path.expand(tool[["perceval"]])
-  mbox_path <- path.expand(conf[["mailing_list"]][["mbox"]])
-
-  project_mbox <- parse_mbox(perceval_path,mbox_path)
+  mbox_file_path <- path.expand(conf[["mailing_list"]][["mod_mbox"]][["project_key_1"]][["mbox_file_path"]])
 
-  cli_alert_success(paste0("Tabulated mailing list was saved at: ",save_path))
+  project_mbox <- parse_mbox(perceval_path, mbox_file_path)
 
-  data.table::fwrite(project_mbox,save_path)
-}else if(arguments[["download"]] & arguments[["modmbox"]] & arguments[["help"]]){
-  cli_alert_info("Saves a mailing list archive from mod_mbox as a .mbox file
-                 using download_mod_mbox().")
-}else if(arguments[["download"]] & arguments[["modmbox"]]){
+  data.table::fwrite(project_mbox, save_path)
+  cli::cli_alert_success(paste0("Tabulated mailing list was saved at: ", save_path))
 
+} else if (arguments[["download"]] & arguments[["modmbox"]] & arguments[["help"]]) {
+  cli::cli_alert_info("Downloads mailing list archives from mod_mbox using download_mod_mbox().")
+} else if (arguments[["download"]] & arguments[["modmbox"]]) {
 
   conf_path <- arguments[["<project_conf.yml>"]]
-  save_path <- arguments[["<save_file_name_path>"]]
-  conf <- yaml::read_yaml(conf_path)
+  start_year_month <- arguments[["<start_year_month>"]]
+  end_year_month <- arguments[["<end_year_month>"]]
+  save_folder_path <- arguments[["<save_folder_path>"]]
 
-  mod_mbox_url <- conf[["mailing_list"]][["domain"]]
-  mailing_list <- conf[["mailing_list"]][["list_key"]][1]
-
-  start_year <- arguments[["<start_year>"]]
-  end_year <- arguments[["<end_year>"]]
+  conf <- yaml::read_yaml(conf_path)
+  mailing_list <- conf[["mailing_list"]][["mod_mbox"]][["project_key_1"]][["mailing_list"]]
```

Comment:
get on all of these hardcoded paths. it will cascade to the execs for the same reason they cascade to the notebooks.

Reply:
@RavenMarQ fyi for your task

Reply:
The corresponding getter functions for these two LOC (66 and 67) are:
`conf <- parse_config(conf_path)`
`mailing_list <- get_mbox_domain(conf, "project_key_1")`

Reply:
Thank you @beydlern 
@carlosparadis I now have it using the getter functions, but the directory is not created if it does not exist, i.e. I have to manually create the folders for ../../rawdata/helix/mod_mbox/save_mbox_mail/ before I can call the download functions. I recall you telling me to not check for the existence of directories in my mail functions, as this would be done somewhere else in the configuration. Should I be doing something about this?